Herebyfile.mjs

```diff
@@ -687,3 +702,532 @@ export class Debouncer {
+                return "arm";
+            case "arm64":
+                return "arm64";
+            default:
+                throw new Error(`Unsupported ARCH: ${arch}`);
+        }
+    }
+});
+
+export const buildNativePreviewPackages = task({
+    name: "native-preview:build-packages",
+    hiddenFromTaskList: true,
+    run: async () => {
+        await rimraf(builtNpm);
+
+        const platforms = nativePreviewPlatforms();
+
+        const inputDir = "./_packages/native-preview";
+
+        const inputPackageJson = JSON.parse(fs.readFileSync(path.join(inputDir, "package.json"), "utf8"));
+        inputPackageJson.version = getVersion();
+        delete inputPackageJson.private;
+
+        const { stdout: gitHead } = await $pipe`git rev-parse HEAD`;
+        inputPackageJson.gitHead = gitHead;
+
+        const mainPackage = {
+            ...inputPackageJson,
+            optionalDependencies: Object.fromEntries(platforms.map(p => [p.npmPackageName, getVersion()])),
+        };
+
+        const mainPackageDir = mainNativePreviewPackage.npmDir;
+
+        await fs.promises.mkdir(mainPackageDir, { recursive: true });
+
+        await cpWithoutNodeModulesOrTsconfig(inputDir, mainPackageDir);
+
+        await fs.promises.writeFile(path.join(mainPackageDir, "package.json"), JSON.stringify(mainPackage, undefined, 4));
+        await fs.promises.copyFile("LICENSE", path.join(mainPackageDir, "LICENSE"));
+        // No NOTICE.txt here; does not ship the binary or libs. If this changes, we should add it.
+
+        let ldflags = "-ldflags=-s -w";
+        if (options.setPrerelease) {
+            ldflags += ` -X github.com/microsoft/typescript-go/internal/core.version=${getVersion()}`;
+        }
+        const extraFlags = ["-trimpath", ldflags];
+
+        const buildLimit = pLimit(os.availableParallelism());
+
+        await Promise.all(platforms.map(async ({ npmDir, npmPackageName, nodeOs, nodeArch, goos, goarch }) => {
+            const packageJson = {
+                ...inputPackageJson,
+                bin: undefined,
+                imports: undefined,
+                name: npmPackageName,
+                os: [nodeOs],
+                cpu: [nodeArch],
+                exports: {
+                    "./package.json": "./package.json",
+                },
+            };
+
+            const out = path.join(npmDir, "lib");
+            await fs.promises.mkdir(out, { recursive: true });
+            await fs.promises.writeFile(path.join(npmDir, "package.json"), JSON.stringify(packageJson, undefined, 4));
+            await fs.promises.copyFile("LICENSE", path.join(npmDir, "LICENSE"));
+            await fs.promises.copyFile("NOTICE.txt", path.join(npmDir, "NOTICE.txt"));
+
+            const readme = [
+                `# \`${npmPackageName}\``,
+                "",
+                `This package provides ${nodeOs}-${nodeArch} support for [${packageJson.name}](https://www.npmjs.com/package/${packageJson.name}).`,
+            ];

```

Comment:
This is a placeholder; I don't think we need any more than this but let me know.